pkg/cluster/internal/create/create.go

```diff
@@ -86,8 +86,16 @@ func Cluster(ctx *context.Context, cfg *config.Cluster, opts *Options) error {
 	}
 	if opts.SetupKubernetes {
 		actionsToRun = append(actionsToRun,
-			kubeadminit.NewAction(),                   // run kubeadm init
-			installcni.NewAction(),                    // install CNI
+			kubeadminit.NewAction(), // run kubeadm init
+		)
+		// this step might be skipped, but is next after init
+		if !cfg.Networking.DisableDefaultCNI {
+			actionsToRun = append(actionsToRun,
+				installcni.NewAction(), // install CNI
```

Comment:
we could alternatively guard against this inside the action, I'm not sure this detail is important currently though as both are wholly internal to kind and should have roughly the same effect, but that _might_ be a better refactor in the future.